packages/core/database/src/utils/identifiers/shortener.ts

```diff
@@ -56,12 +66,38 @@ export function createHash(data: string, len: number): string {
     throw new Error(`createHash length must be a positive integer, received ${len}`);
   }
 
-  // TODO: shake256 is based on SHA-3 and is slow, we don't care about cryptographically secure, only uniqueness and speed
-  //       investigate alternatives before releasing this. But it is only run on startup, so it should be fine.
   const hash = crypto.createHash('shake256', { outputLength: Math.ceil(len / 2) }).update(data);
   return hash.digest('hex').substring(0, len);
 }
 
+// TODO: the entire identifiers lib could be a factory and this map could be stored internally instead of globally
+// We need to be able to find the full-length name for any shortened name, primarily for migration purposes
+// Therefore we store every name that passes through so we can retrieve the original later
+const nameMap = new Map<string, string>();
```

Comment:
A few thoughts

- We couble be able to remove that map in a later release as it is only necessary to run the migration 🤔 
- Should we clear the map at the end of the db init phase

Reply:
Yeah, we can remove big parts of this in a later release as long as we require migrating to 5.0.0 (or whatever) first.

I think leaving the map could be potentially useful, it doesn't use much memory. But I also don't mind removing it, the use cases for it are pretty limited.

Reply:
Yeah if we see a need to remove it we can but not worth planning for it :)